spec/controllers/transfers_controller_spec.rb

```diff
@@ -207,5 +207,22 @@
         end
       end
     end
+
+    context 'with invalid params' do
+      let(:user) { member_giver.user }
+      let(:referer) { "/transfers/new?destination_account_id=#{member_taker.account.id}" }
+
+      before do
+        request.env["HTTP_REFERER"] = referer
```

Comment:
I'd restore its value afterward so as not to mess up with test suite's state and avoid future headaches.

Reply:
@sauloperez do you mean something like this?

```diff
diff --git a/spec/controllers/transfers_controller_spec.rb b/spec/controllers/transfers_controller_spec.rb
index c075091..5aed2e7 100644
--- a/spec/controllers/transfers_controller_spec.rb
+++ b/spec/controllers/transfers_controller_spec.rb
@@ -213,9 +213,14 @@ RSpec.describe TransfersController do
       let(:referer) { "/transfers/new?destination_account_id=#{member_taker.account.id}" }
 
       before do
+        @prev_referer = request.env["HTTP_REFERER"]
         request.env["HTTP_REFERER"] = referer
       end
 
+      after do
+        request.env["HTTP_REFERER"] = @prev_referer
+      end
+
```

Reply:
something alike, but I believe [around](https://relishapp.com/rspec/rspec-core/v/3-7/docs/hooks/around-hooks) is better suited for that.

Reply:
I tried it first with `around`, but didn't get it working (you can't share @ivars, the `request` object returns nil at that point). 

To be honest, I have never used this type of reset the referer header, in larger suites. It didn't sound to me having problems with this, isn't reset automatically on each request? am I missing something 😅?

Reply:
Fuuu right it's the request object so pointless comment :see_no_evil: